### ISSUE TITLE
Async operations inside Auth0Lock constructor

### ISSUE THREAD

**Issue by rodrijuarez** [auth0/lock#1163]
I've been using the Auth0 Lock widget and I've encountered this problem related to the autofill inside Chrome (any version). This is probably related with async operations inside the constructor of Auth0Lock, is there any way to check wether all the operations inside the constructor of Auth0Lock were finished correctly? Given that it doesn't return a promise nor have option for a callback.

 Maybe I'm making a mistake saying that there aren't async operations, I'm just guiding by these lines of code:

https://github.com/auth0/lock/blob/e3dbc5258084b00c4671e9f47c750bef6a3110da/src/core/remote_data.js

Basically, this causes the autofill of Chrome not display correctly whenever I try to call the show function from the Auth0Lock object returned just after the creation of Auth0Lock object, as you can see in the two images below.

Lock version: v10.17.0

Without setTimeout

![screen shot 2017-11-06 at 17 12 37](https://user-images.githubusercontent.com/12787254/32450934-dfc76928-c315-11e7-979f-21bcbb5fb0b8.png)

With setTimeout 500ms (Expected Result)

![screen shot 2017-11-06 at 17 13 24](https://user-images.githubusercontent.com/12787254/32450935-dfe6da24-c315-11e7-8baf-88b5908f87ec.png)

**Comment by luisrudge** [comment]
I'm not sure what's going on. We have to fetch your client's settings when Auth0Lock is instantiated, so we do an ajax call. Can you clarify what's going on with your project? Are you saying that chrome is only auto filling your inputs if you use a setTimeout?

**Comment by rodrijuarez** [comment]
@luisrudge it seems to me that the problem is related to that, is there any way to expose any kind of event for when this load finishes? Do you guys stop the rendering till this requests are done? 

Works: 

```
   setTimeout(() => this.lockWidget.show({ initialScreen: <any>initialScreen }), 500);
```

Doesn't work:

```
   this.lockWidget.show({ initialScreen: <any>initialScreen })
```

I don't know how you guys handle the case where the constructor is still fetching data when the show is emitted. I tried to go a little bit through the code but it's kind of difficult without knowing the codebase.

**Comment by luisrudge** [comment]
hmm. interesting. We don't emit an event, but we could emit one. I don't think this is the best solution though. I'm also not sure what would be the best thing here. We render the widget before the request is done, but with a loading animation instead of a form. The question is: why does chrome autofills when there is no loading animation presented (the first render is already the form), but doesn't autofill if the form is "added" after the loading animation.  That's very intriguing. I'll have to separate a time to look a this in the future. We're kinda in the middle of a big release and I can't allocate time to look at this. If you do have the time, I'll help you with any questions you have.

**Comment by rodrijuarez** [comment]
@luisrudge it'd be nice to have the emission of the event or something so we can solve this issue at least without the usage of setTimeout or any other thing. I'd try forking the codebase but it's too much of a mess and it'd require a lot of time for such a small thing. 

 Good luck with the release!

**Comment by luisrudge** [comment]
Hi! I'm sorry about the copy/paste message, but I'm cleaning up some stale issues. 🗑 
It's been a while since this issue was opened. Are you still having this issue with the latest version? If so, please reopen it with some repro steps so we can help!
